### PR TITLE
[PIN-1295] migrating POST ...{descriptorId}/suspend from catalog-process

### PR DESCRIPTION
**Pull request by bloodsky**

### CODE REVIEW

File: src/main/resources/interface-specification.yml [hunk 1]
```diff
@@ -1749,6 +1749,45 @@ paths:
...
+        - eservices
+      summary: Suspend the selected descriptor.
+      operationId: suspendDescriptor
+      parameters:
+        - name: eServiceId
+          in: path
+          description: the eservice id
+          required: true
+          schema:
+            type: string
```
Comment (galales) [inline src/main/resources/interface-specification.yml:2051]:
```suggestion
            type: string
            format: uuid
```

File: src/main/resources/interface-specification.yml [hunk 2]
```diff
@@ -1749,6 +1749,45 @@ paths:
...
+        - name: descriptorId
+          in: path
+          description: the descriptor Id
+          required: true
```
Comment (galales) [inline src/main/resources/interface-specification.yml:2056]:
```suggestion
          required: true
          format: uuid
```

File: src/main/resources/interface-specification.yml [hunk 3]
```diff
@@ -1749,6 +1749,45 @@ paths:
...
+          content:
+            application/problem+json:
+              schema:
+                $ref: '#/components/schemas/Problem'
+        '404':
+          description: Not found
+          content:
+            application/problem+json:
+              schema:
+                $ref: '#/components/schemas/Problem'
```
Comment (galales) [inline src/main/resources/interface-specification.yml]: missing rate limiting headers

File: src/main/scala/it/pagopa/interop/backendforfrontend/service/impl/CatalogProcessServiceImpl.scala [hunk 4]
```diff
@@ -64,4 +64,16 @@ class CatalogProcessServiceImpl(catalogProcessUrl: String, blockingEc: Execution
       invoker.invoke(request, s"Retrieving EService for $eServiceId from Catalog Process")
     }
 
+  override def suspendDescriptor(eServiceId: String, descriptorId: String)(implicit
+    contexts: Seq[(String, String)]
+  ): Future[Unit] = withHeaders { (bearerToken, correlationId, ip) =>
+    val request: ApiRequest[Unit] =
+      api.suspendDescriptor(
+        xCorrelationId = correlationId,
+        eServiceId = eServiceId.toString,
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/service/impl/CatalogProcessServiceImpl.scala]: eServiceId is already a String
```suggestion
        eServiceId = eServiceId,
```

**Review by galales — CHANGES_REQUESTED** [review]

**Review by beetlecrunch — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/backendforfrontend/service/impl/CatalogProcessServiceImpl.scala [hunk 5]
```diff
@@ -64,4 +64,16 @@ class CatalogProcessServiceImpl(catalogProcessUrl: String, blockingEc: Execution
...
+    contexts: Seq[(String, String)]
+  ): Future[Unit] = withHeaders { (bearerToken, correlationId, ip) =>
+    val request: ApiRequest[Unit] =
+      api.suspendDescriptor(
+        xCorrelationId = correlationId,
+        eServiceId = eServiceId,
+        descriptorId = descriptorId,
+        xForwardedFor = ip
+      )(BearerToken(bearerToken))
+    invoker.invoke(request, s"Retrieving EService for $eServiceId from Catalog Process")
```
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/backendforfrontend/service/impl/CatalogProcessServiceImpl.scala:126]: ditto with the other PR: IMHO a bit of error handling won't hurt, but let's discuss it together

**Review by TonioGela — CHANGES_REQUESTED** [review]

File: src/main/scala/it/pagopa/interop/backendforfrontend/service/impl/CatalogProcessServiceImpl.scala (hunk 5 above)
Comment (galales) [reply src/main/scala/it/pagopa/interop/backendforfrontend/service/impl/CatalogProcessServiceImpl.scala:126]: [same here](https://github.com/pagopa/interop-be-backend-for-frontend/pull/85#discussion_r1094265963)

**Review by TonioGela — APPROVED** [review]

**Review by galales — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/backendforfrontend/service/CatalogProcessService.scala [hunk 6]
```diff
@@ -7,28 +7,32 @@ import scala.concurrent.Future
...
 
-  def getEServices(
-    name: Option[String] = None,
-    eServicesIds: Seq[UUID],
-    producersIds: Seq[UUID],
-    states: Seq[EServiceDescriptorState],
-    offset: Int,
-    limit: Int
-  )(implicit contexts: Seq[(String, String)]): Future[EServices]
+  def createEService(eServiceSeed: EServiceSeed)(implicit contexts: Seq[(String, String)]): Future[EService]
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/service/CatalogProcessService.scala]: please do not move code 🙏 
can you place them back?

PS: this file needs formatting

**Review by galales — CHANGES_REQUESTED** [review]

**Review by galales — APPROVED** [review]

**Pull request merged**